jobs: TestLoseLeaseDuringExecution failed

jobs.TestLoseLeaseDuringExecution [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3097174&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3097174&tab=artifacts#/) on master @ [83b68660960d0b094943eb96549170fbe2c0df37](https://github.com/cockroachdb/cockroach/commits/83b68660960d0b094943eb96549170fbe2c0df37):


```
=== RUN   TestLoseLeaseDuringExecution
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution826201177
    test_log_scope.go:74: use -show-logs to present logs inline
    jobs_test.go:2760: 
        	Error Trace:	jobs_test.go:2760
        	Error:      	Expect "<nil>" to match "expected session '\w+' but found NULL"
        	Test:       	TestLoseLeaseDuringExecution
E210618 16:50:53.879715 87264 jobs/registry.go:648  [-] 1  error processing claimed jobs: could not query for claimed jobs: context canceled
E210618 16:50:53.881443 87262 jobs/registry.go:615  [-] 2  error expiring job sessions: expire-sessions: context canceled
E210618 16:50:53.882197 87262 jobs/registry.go:623  [-] 3  failed to serve pause and cancel requests: context canceled
    panic.go:636: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution826201177
--- FAIL: TestLoseLeaseDuringExecution (1.41s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestLoseLeaseDuringExecution PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/sql-schema ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLoseLeaseDuringExecution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

jobs.TestLoseLeaseDuringExecution [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3098431&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3098431&tab=artifacts#/) on master @ [69308cce3bb7e660908cb3e2724eedd271ce5585](https://github.com/cockroachdb/cockroach/commits/69308cce3bb7e660908cb3e2724eedd271ce5585):


```
=== RUN   TestLoseLeaseDuringExecution
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution014865227
    test_log_scope.go:74: use -show-logs to present logs inline
    jobs_test.go:2760: 
        	Error Trace:	jobs_test.go:2760
        	Error:      	Expect "<nil>" to match "expected session '\w+' but found NULL"
        	Test:       	TestLoseLeaseDuringExecution
E210619 06:23:54.279647 80858 jobs/registry.go:648  [-] 1  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution014865227
--- FAIL: TestLoseLeaseDuringExecution (0.14s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestLoseLeaseDuringExecution PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/sql-schema ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLoseLeaseDuringExecution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

jobs.TestLoseLeaseDuringExecution [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3100097&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3100097&tab=artifacts#/) on master @ [0e747e4ac3630a9b41f2eccb7ace0e2d17258ae6](https://github.com/cockroachdb/cockroach/commits/0e747e4ac3630a9b41f2eccb7ace0e2d17258ae6):


```
=== RUN   TestLoseLeaseDuringExecution
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution404860533
    test_log_scope.go:74: use -show-logs to present logs inline
    jobs_test.go:2760: 
        	Error Trace:	jobs_test.go:2760
        	Error:      	Expect "<nil>" to match "expected session '\w+' but found NULL"
        	Test:       	TestLoseLeaseDuringExecution
E210621 05:59:28.910635 80967 jobs/registry.go:623  [-] 1  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution404860533
--- FAIL: TestLoseLeaseDuringExecution (0.23s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestLoseLeaseDuringExecution PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/sql-schema ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLoseLeaseDuringExecution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

jobs.TestLoseLeaseDuringExecution [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3101832&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3101832&tab=artifacts#/) on master @ [d9769db8d68b76df7c0d806362047d9d171ef8b7](https://github.com/cockroachdb/cockroach/commits/d9769db8d68b76df7c0d806362047d9d171ef8b7):


```
=== RUN   TestLoseLeaseDuringExecution
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution271947319
    test_log_scope.go:74: use -show-logs to present logs inline
    jobs_test.go:2760: 
        	Error Trace:	jobs_test.go:2760
        	Error:      	Expect "<nil>" to match "expected session '\w+' but found NULL"
        	Test:       	TestLoseLeaseDuringExecution
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLoseLeaseDuringExecution271947319
--- FAIL: TestLoseLeaseDuringExecution (0.50s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestLoseLeaseDuringExecution PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/sql-schema ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLoseLeaseDuringExecution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -2718,7 +2718,7 @@ func TestLoseLeaseDuringExecution(t *testing.T) {
 	defer jobs.ResetConstructors()()
 
 	// Disable the loops from messing with the job execution.
-	knobs := base.TestingKnobs{JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals()}
+	knobs := base.TestingKnobs{JobsTestingKnobs: jobs.NewTestingKnobsWithIntervals(time.Hour, time.Hour)}
 
 	ctx := context.Background()
 

